### PR TITLE
Add example python extension

### PR DESCRIPTION
**Pull request by benjie**
I've not written Python in many years and forgot about `__init__.py` etc, so I figured this walkthrough on how to get started with writing a Python plugin for Albert would be a nice addition to the docs. Thanks for all your hard work :raised_hands:

### CODE REVIEW

**Comment by ManuelSchneid3r** [comment]
ty

**Comment by ManuelSchneid3r** [comment]
sorry benji i had to partially revoke your work for several reasons:

* redundancy. the information is basically repeating the stuff above
* to specific, without mentioning it.  the paths are linux only
* naming. the header of the document explicitly set a terminology.
* no code in docs please. also there is a dedicated extension to get started already.
* quality. saw some typos 

i took just a quick look before merging. should have watch more closely. however I see that the python ext doc is absolutely not readable. i rewrote it. 

anyway thank you for your help. and directing me to the the grotty docs i wrote.

**Comment by benjie** [comment]
> to specific, without mentioning it. the paths are linux only

I didn't realise Albert was cross-platform!

> also there is a dedicated extension to get started already.

There's one in C++ for sure; I thought it'd be handy to have one in Python too.

> no code in docs please. 

"no code in docs" is an interesting rule, out of interest (as a fellow open source maintainer) what was the motivation for that?

> quality. saw some typos

It would be helpful to me if you could point these out explicitly.

>  however I see that the python ext doc is absolutely not readable. i rewrote it.

The getting started section and link to the existing extensions is really helpful; it took me quite a while to figure out how to get started before.

It would be nice to link to a very simple "example" plugin (such as the UUID one I wrote) so that people who just need to do something really simple don't need to understand the threading APIs in Python to grok the example code. If you'd like me to submit the UUID extension to the python extensions repo I'd be happy to do so.

**Comment by benjie** [comment]
Off topic, I notice you have GitHub Sponsors now; would you like me to move my sponsorship over from Patreon?

File: src/_docs/extensions/python.md [hunk 1]
```diff
@@ -183,3 +183,88 @@ Ids are guaranteed to be unique. This means that if several of those paths conta
...
+4. Open Albert settings and navigate to Extensions > Python
+5. If the "UUID" option is not listed, fully quit and re-open Albert
+6. Check the checkbox next to UUID, the light should turn green
+7. Use the extension by triggering albert and typing 'uuid'
+
+If the light turns red, there was an error loading your extension - hover over the name to view the details.
+
+It can be useful for debugging to run Albert from a terminal so that you can see the output from your script.
+
+Though reloading Albert is may be necessary to recognize that a python script is added, it is not required to detect changes to the script.
```
Comment (ManuelSchneid3r) [inline src/_docs/extensions/python.md:203]: here typo

**Comment by ManuelSchneid3r** [comment]
> "no code in docs" is an interesting rule, out of interest (as a fellow open source maintainer) what was the motivation for that?

because I almost certainly forget to update this example whenever the extension api finally changes. I dont like the fact that there are consistancy dependencies over decoupled places. I hope to get to the core soon because I want to add READMEs which should get sourced by jekyll. this way code and docs stay on the same version.

> It would be helpful to me if you could point these out explicitly.

i commented it. okay I have to admit plural was exaggerated. however there is at least one 😄 but actually thats not much of a problem .

> I thought it'd be handy to have one in Python too.

there is one. it has just an unfortunate name. the api test extension should give you a quick overview over most of the things you can do in python.

> I didn't realise Albert was cross-platform!

it is not yet. however most of the code is platform agnostic and since my main system is a macbook I plan to port it to mac soon. I hope this to boost my dev time for albert since I my linux system meanwhile actually exists only for albert dev. and most of my time i am not close to it.

> It would be nice to link to a very simple "example" plugin (such as the UUID one I wrote) so that people who just need to do something really simple don't need to understand the threading APIs

i dont get this one. threading? there is no threading. (sadly. the PIL cost my dozens of dozens of hours since I really liked to see python be MT like the core too) I think the api ext is really not that complicated if you are capable of writing basic python. have you checked it?

**Comment by ManuelSchneid3r** [comment]
> Off topic, I notice you have GitHub Sponsors now; would you like me to move my sponsorship over from Patreon?

sure. I have no experience with it, but at least github said they will covers the "transaction cost". which sounds like no fees to me. thank you for supporting. I hope you enjoy albert.

File: src/_docs/extensions/python.md (hunk 1 above)
Comment (benjie) [reply src/_docs/extensions/python.md:203]: Ah; I accidentally left the word “is” in; got it 👍

**Comment by benjie** [comment]
I meant like the threading in https://github.com/albertlauncher/python/blob/dev/unicode_emoji/__init__.py; calling out api_test as the example/simple plugin would be helpful (I was expecting it to be a complex HTTP API tester integration of some kind, so skipped over it for something I expected to be simple: emoji). There’s value in having a tiny plugin people can build up from; not everyone is proficient in Python and C. Anyway the getting started is a big improvement so I think we’ve made good progress here.

**Pull request merged**